api/src/main/java/org/openmrs/module/fhir/api/impl/DiagnosticReportServiceImpl.java

```diff
@@ -68,16 +96,23 @@ public void setDao(FHIRDAO dao) {
 	}
 
 	@Override
-	public DiagnosticReport getDiagnosticReport(String id) {
-		// Find Diagnostic Report (Encounter) in OpenMRS database
-		EncounterService encounterService = Context.getEncounterService();
-		Encounter omrsDiagnosticReport = encounterService.getEncounterByUuid(id);
-		// Get corresponding Handler
-		String handlerName = omrsDiagnosticReport.getEncounterType().getName();
-
-		return FHIRDiagnosticReportUtil.getFHIRDiagnosticReport(id, getHandler(handlerName));
+	public DiagnosticReport getDiagnosticReport(final String accessionNumber) {
+		loadHandlerMap();
+		List<Order> orders = dao.getOrdersByAccessionNumber(accessionNumber);
+		if (orders != null && !orders.isEmpty()) {
```

Comment:
I am just curious.
What the difference between **orders != null and  !orders.isEmpty()** ?

Reply:
@corneliouzbett If `getOrdersByAccessionNumber()` returned null, we'd get a NullPointerException when trying to call `orders.isEmpty()`. That said, I dug into a bit and I think Hibernate won't return null here, so I've dropped that check.